core/src/main/scala/org/dbpedia/extraction/mappings/wikidata/WikidataR2RExtractor.scala

```diff
@@ -219,18 +232,45 @@ class WikidataR2RExtractor(
 
     value match {
       case v: ItemIdValue => {
+        val wikidataItem = WikidataUtil.getItemId(v)
         val valueTitle = "wikidata:" + WikidataUtil.getItemId(v)
-        context.ontology.wikidataClassesMap.foreach({ map =>
-          if (map._1.matches(valueTitle)) {
-            classes ++= map._2
+        try {
+          val getOntologyKey = itemMap.get(wikidataItem)
+          getOntologyKey match {
+            case Some(key) =>
+              classes++=Set(context.ontology.classes(key))
+            case _ =>
           }
-        })
+        } catch {
+          case ex:Exception =>
+        }
+          context.ontology.wikidataClassesMap.foreach({ map =>
+            if (map._1.matches(valueTitle)) {
+              classes ++= map._2
+            }
+          })
+
       }
       case _ =>
     }
     classes
   }
 
+  private def readJson(fileName:String): Map[String,String] = {
```

Comment:
generalize this function that appends the ontology mappings and only use the itemMap in the code
(better rename itemMap to classMappings)